src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/JwtVerifier.java

```diff
@@ -130,8 +131,10 @@ private void validateRequiredAudienceAndIssuer(JWTClaimsSet claims) throws BadJW
         String audience = claims.getAudience().stream().findFirst().orElse("");
         String issuer = claims.getIssuer();
 
-        if (!Strings.isNullOrEmpty(requiredAudience) && !requiredAudience.equals(audience)) {
-            throw new BadJWTException("Invalid audience");
+        if (requiredAudience.length >0) {
+            if (!Arrays.asList(requiredAudience).contains(audience)) {
+                throw new BadJWTException("Invalid audience");
+            }
```

Comment:
Thoughts on this, once required_audience is set to string list:
```suggestion
        if (requiredAudience.isEmpty() || !requiredAudience.contains(audience)) {
                throw new BadJWTException("Invalid audience");
            }
```